src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -236,6 +296,47 @@ impl<'tcx> Loan<'tcx> {
     pub fn loan_path(&self) -> Rc<LoanPath<'tcx>> {
         self.loan_path.clone()
     }
+
+    pub fn kind(&self) -> ty::BorrowKind {
+        self.kind
+    }
+
+    pub fn span(&self) -> Span {
+        self.span
+    }
+
+    pub fn span_in_def(&self, def_id: DefId, tcx: TyCtxt) -> Option<Span> {
```

Comment:
Any ideas for a better name? This is to generate the span (if it exists) for the "scope" of the value pointed to by the `DefId`. Should I add comments adding more details?